File: baekjoon/자료 구조/[10866]덱.py
from collections import deque
import sys

dq = deque()

def empty():
    if len(dq) == 0:
        return 1
    else:
        return 0

n = int(input())
        
for i in range(n):
    cmd = sys.stdin.readline().split()
    
    if cmd[0] == "push_front":
        dq.appendleft(cmd[1])
    if cmd[0] == "push_back":
        dq.append(cmd[1])
    if cmd[0] == "pop_front":
        if empty() == 1:
            print(-1)
        else:
            print(dq.popleft())
    if cmd[0] == "pop_back":
        if empty() == 1:
            print(-1)
        else:
            print(dq.pop())
    if cmd[0] == "size":
        print(len(dq))
    if cmd[0] == "empty":
        print(empty())
    if cmd[0] == "front":
        if empty() == 1:
            print(-1)
        else: 
            print(dq[0])
    if cmd[0] == "back":
        if empty() == 1:
            print(-1)
        else:
            print(dq[-1])

File: baekjoon/자료 구조/[10866]덱.cpp
#include <iostream>
#include <deque>
using namespace std;

int main() {
    int n;
    int num;
    string cmd;
    deque<int> dq;

    cin >> n;

    for (int i = 0; i < n; i++) {
        cin >> cmd;

        if (cmd == "push_front") {
            cin >> num;
            dq.push_front(num);
        }
        if (cmd == "push_back") {
            cin >> num;
            dq.push_back(num);
        }
        if (cmd == "pop_front") {
            if (!dq.empty()) { 
                cout << dq.front() << "\n";
                dq.pop_front();
            }
            else cout << -1 << "\n";
        }
        if (cmd == "pop_back") {
            if (!dq.empty()) {
                cout << dq.back() << "\n";
                dq.pop_back();
            }
            else cout << -1 << "\n";
        }
        if (cmd == "size") cout << dq.size() << "\n";
        if (cmd == "empty") cout << dq.empty() << "\n";
        if (cmd == "front") {
            if (!dq.empty()) cout << dq.front() << "\n";
            else cout << -1 << "\n";
        }
        if (cmd == "back") {
            if (!dq.empty()) cout << dq.back() << "\n";
            else cout << -1 << "\n";
        }

    }
}